packages/devtools_app/test/cpu_profiler/cpu_profiler_controller_test.dart

```diff
@@ -104,37 +114,54 @@ void main() {
       expect(filteredNativeFrames, isEmpty);
     });
 
-    test('generateToggleFilterSuffix', () {
-      for (final toggleFilter in controller.toggleFilters) {
-        toggleFilter.enabled.value = false;
-      }
-      expect(controller.generateToggleFilterSuffix(), equals(''));
-
-      controller.toggleFilters[0].enabled.value = true;
-      expect(
-        controller.generateToggleFilterSuffix(),
-        equals('Hide Native code'),
+    test('filters data by query filter', () async {
+      // [startMicros] and [extentMicros] are arbitrary for testing.
+      await controller.pullAndProcessProfile(
+        startMicros: 0,
+        extentMicros: 100,
+        processId: 'test',
       );
-
-      controller.toggleFilters[1].enabled.value = true;
+      final originalData = controller.cpuProfileStore.lookupProfile(
+        label: CpuProfilerController.userTagNone,
+      )!;
       expect(
-        controller.generateToggleFilterSuffix(),
-        equals('Hide Native code,Hide core Dart libraries'),
+        originalData.functionProfile.stackFrames.values.length,
+        equals(17),
       );
 
-      controller.toggleFilters[2].enabled.value = true;
-      expect(
-        controller.generateToggleFilterSuffix(),
-        equals(
-          'Hide Native code,Hide core Dart libraries,Hide core Flutter libraries',
-        ),
-      );
+      // At this point, data is filtered by the default toggle filter values.
+      var filteredData = controller.dataNotifier.value!;
+      expect(filteredData.stackFrames.values.length, equals(12));
```

Comment:
Ubernit: the `equals(...)` matcher is the default, so it's not necessary. Feel free to remove it if you'd like.

Reply:
ack. this is a big clean up, so I filed https://github.com/flutter/devtools/issues/5210 to track.